packages/amplify-graphql-relational-transformer/src/resolvers.ts

```diff
@@ -42,20 +93,42 @@ export const updateTableForConnection = (config: HasManyDirectiveConfiguration,
   const partitionKeyType = respectPrimaryKeyAttributesOnConnectionField
     ? attributeTypeFromType(getObjectPrimaryKey(object).type, ctx)
     : 'S';
-  const sortKeyAttributeDefinitions = respectPrimaryKeyAttributesOnConnectionField
-    ? getConnectedSortKeyAttributeDefinitionsForImplicitHasManyObject(ctx, object, field)
+  const sortKey = respectPrimaryKeyAttributesOnConnectionField
+    ? getConnectedSortKeyAttributeDefinitionsForImplicitHasManyObject(ctx, object, field, getSortKeyFields(ctx, object))
     : undefined;
+
+    addGlobalSecondaryIndex(table, {
+      indexName: indexName,
+      partitionKey: { name: partitionKeyName, type: partitionKeyType },
+      sortKey: sortKey,
+      ctx: ctx,
+      relatedTypeName: relatedType.name.value,
+    });
+};
+
+const addGlobalSecondaryIndex = (
+  table: any,
+  props: {
+    indexName: string,
+    partitionKey: KeyAttributeDefinition,
+    sortKey: KeyAttributeDefinition | undefined,
+    ctx: TransformerContextProvider,
+    relatedTypeName: string,
```

Comment:
Can you double-check that this is working as expected with GSIs added to the related model? It looks like the `relatedTypeName` is used to override the index name in CFN, but I'm a bit fuzzy on why we need to. And since we're already creating the index properly I'm unclear what the correct behavior should be.

Reply:
`relatedTypeName` is being used within `overrideIndexAtCfnLevel` to determine the data source strategy for that model, which is then used to determine how to override the index. It's not being used to override the index name in CFN.

This existing comment may provide some clarity.
https://github.com/aws-amplify/amplify-category-api/pull/2362/files/0eeb2cf93594f7d9ee6e872d5244a5274edb1719#diff-7e54ffdc31c20805d81c17a3e63863a1553b9911a6a00daf1e21f027f2995d84L65-L67
```
  // At the L2 level, the CDK does not handle the way Amplify sets GSI read and write capacity
  // very well. At the L1 level, the CDK does not create the correct IAM policy for accessing the
  // GSI. To get around these issues, keep the L1 and L2 GSI list in sync.
```